### ISSUE TITLE
Not working on spigot 1.8 anymore

### ISSUE THREAD

**Issue by iaxedu** [LilyPad/Bukkit-CompatBungee#1]
http://pastebin.com/m6RkD80E

**Comment by coelho** [comment]
This is due to the ProtocolLib version you're using not being complete. 
Not really something LilyPad can fix.

**Comment by dmulloy2** [comment]
This is an issue with LilyPad.  The payload byte array was replaced with the PacketDataSerializer in the payload classes:

``` java
private String a;
private PacketDataSerializer b; // Replaced byte array
```

**Comment by coelho** [comment]
@dmulloy2

Looking into this further I do not seem to be able to get a PacketDataSerializer (or any of ProtocolLib's wrappers for it) from PacketContainer. Correct me if I'm wrong?

**Comment by dmulloy2** [comment]
There isn't a wrapper for PacketDataSerializer, but something like this would work:

``` java
PacketContainer packet = event.getPacket();

// Obtain the data serializer
Class<?> dataSerializerClass = MinecraftReflection.getPacketDataSerializerClass();
StructureModifier<?> dataSerializers = packet.getSpecificModifier(dataSerializerClass);
Object dataSerializer = dataSerializers.read(0);

// Obtain the methods we will need
// You can probably make these static
FuzzyReflection fuzzy = FuzzyReflection.fromClass(dataSerializerClass);
MethodAccessor readableBytes = Accessors.getMethodAccessor(fuzzy.getMethodByParameters("readableBytes", int.class, new Class<?>[0]));
MethodAccessor readBytes = Accessors.getMethodAccessor(fuzzy.getMethodByParameters("readBytes", byte[].class, new Class<?>[] { int.class }));

int i = (int) readableBytes.invoke(dataSerializer);
byte[] bytes = (byte[]) readBytes.invoke(dataSerializer, i);
// Do stuff
```

**Comment by coelho** [comment]
Resolved by PR #2
